src/joystick/joystick_manager.cpp

```diff
@@ -0,0 +1,301 @@
+        if (getVal(msg.axes, xbox360mapping.at(btn)) > 0) {                   // If DpadDown is pressed
+            if (!getVal(msg_prev.axes, xbox360mapping.at(btn)) >
+                0) {        // Check whether it was not already pressed before
+                joy.setRobotID((joy.robotID + 1) % 16);                     // If not, increment id
+            }
+        } else if (getVal(msg.axes, xbox360mapping.at(btn)) < 0) {                   // If DpadUp is pressed
+            if (getVal(msg_prev.axes, xbox360mapping.at(btn)) ==
+                -0) {       // Check whether it was not already pressed before
+                joy.setRobotID((joy.robotID + 15) % 16);                     // If not, decrement id
+            }
+        }
+    }
+
+    /* ==== Rotate kicker (Geneva Drive) if X is pressed ==== */
+    if(getVal(msg.buttons, xbox360mapping.at(Xbox360Controller::X))) {      // Press X to rotate the Geneva Drive
+        /* ==== Rotate kicker (Geneva Drive)==== */
+        btn = Xbox360Controller::DpadX;
+        if(getVal(msg.axes, xbox360mapping.at(btn)) > 0){                   // If DpadRight is pressed
+            if(getVal(msg_prev.axes, xbox360mapping.at(btn)) == -0) {       // Check whether it was not already pressed before
+                if(joy.genevaState < 5) {                                   // Check whether the state is not already maximum
+                    joy.genevaState++;                                      // Increase geneva state by 1
+                }
+            }
+        }else
+        if(getVal(msg.axes, xbox360mapping.at(btn)) < 0){                   // If DpadLeft is pressed
+            if(getVal(msg_prev.axes, xbox360mapping.at(btn)) == -0) {       // Check whether it was not already pressed before
+                if(joy.genevaState > 1) {                                   // Check whether the state is not already minimal
+                    joy.genevaState--;                                      // Decrease geneva state by 1
+                }
+            }
+        }
+    }
+
+    /* ==== Set rotation offset to current rotation ==== */
+    btn = Xbox360Controller::B;
+    if(getVal(msg.buttons, xbox360mapping.at(btn)) > 0){                // If B is pressed
+        if(!getVal(msg_prev.buttons, xbox360mapping.at(btn)) > 0){      // Check whether it was not already pressed before
+            joy.orientationOffset += joy.orientation;                   // Add current orientation to orientation offset
+            joy.orientation = 0;                                        // Set current orientation to 0
+            ROS_INFO_STREAM(joy.input << " : orientation offset = " << (joy.orientation / (16 * M_PI)));
+        }
+    }
+    /* ================================================== */
+
+}
+
+// ==== Make all robot commands and return them ==== //
+roboteam_msgs::RobotCommand makeRobotCommand(JoyEntry &joy, sensor_msgs::Joy const &msg, sensor_msgs::Joy const &msg_prev) {
```

Comment:
This is outside of the class declaration (?) so you need to send joy again etc. If you put this inside the joyEntry class there is no need to set joy anymore.